crates/storage/provider/src/traits/account.rs

```diff
@@ -42,3 +43,10 @@ pub trait AccountExtReader: Send + Sync {
         range: RangeInclusive<BlockNumber>,
     ) -> Result<BTreeMap<Address, Vec<BlockNumber>>>;
 }
+
+/// AccountChange reader
+#[auto_impl(&, Arc, Box)]
+pub trait AccountChangeReader: Send + Sync {
```

Comment:
minor: is this better named `AccountChangeSetReader` or does that "tie" the trait too closely to the table?

Reply:
`ChangeSet` would be more accurate as we don't have a notion of `Change` but just `ChangeSet`
I would name it as `ChangeSetReader` without Account to be broader

Reply:
Renamed in [a92b817](https://github.com/paradigmxyz/reth/pull/3768/commits/a92b817cb962addac3d28d83da63023a39da98f0)